app.js

```diff
@@ -1,81 +1,141 @@
+  // This points to the window. - JP
+  // };
+
+function counterFunction () {
+  let count = 0;
   // this code is broken. figure out why, and tell us where the closure is when you fix it
-  const changeCount = (value) => {
-    count += value;
-  };
-  return {
-    increment: () => {
-      changeCount(1);
-    },
-    decrement: () => {
-      changeCount(-1);
-    },
-    total: () => {
-      return count;
-    }
+  // **I assume this is supposed to be a counter function.
+  // You just make a function, put count immediately inside of it to make it local. The counting
+  // function is now within the **changeCount function** which holds count so the closure is there.
+  // All of this is boxed up within the counterFunction function. - JP**
+  const changeCount = function () {
+    count = count + 1;
+    console.log(count)
   }
-};
-const counter = counterFunction();
-counter.increment();
-counter.increment();
-counter.increment();
-counter.increment();
-counter.increment();
-console.log(counter.total());
+  return changeCount;
+}
+var increment = counterFunction();
+console.log (increment());
+console.log (increment());
+console.log (increment());
+console.log (increment());
+console.log (increment());
+// sorry but I hate es6 function syntax (for now)
 
 
 // create a constructor function called "Car"
   // car takes an options object as its only argument
   // your options object should have "make", "model", "year" properties on it
   // assign these properties you pass in with options to the constructors 'this' object.
   // add a speak() method to your object that when called will log out the car's make model and year.
-
+  const Car = function (options) {
+    this.make = options.make;
+    this.model = options.model;
+    this.year = options.year;
+    this.speak = function () {
+      console.log(this.make);
+      console.log(this.model);
+      console.log(this.year);
+    };
+  };
 // when you're done un comment the next few lines and run the file here in node `node app.js`.
 
-// const herby = new Car({make: 'Volkswagen', model: 'Beetle', year: '1963'});
-// console.log(herby.speak());
-// const goldfinger = new Car({make: 'Aston Martin', model: 'DB5', year: '1964'});
-// console.log(goldfinger.speak());
-
+const herby = new Car({make: 'Volkswagen', model: 'Beetle', year: '1963'});
+console.log(herby.speak());
 
-// once you get done with this, redo it all using the class keyword and a constructor function. 
+const goldfinger = new Car({make: 'Aston Martin', model: 'DB5', year: '1964'});
+console.log(goldfinger.speak());
 
-// extra credit
 
-// we didn't touch on Recursion in the lecture yet, but you're going to build a recursive function now
+// once you get done with this, redo it all using the class keyword and a constructor function.
+class MyCar {
+constructor(options) {
+  this.make = options.make;
+  this.model = options.model;
+  this.year = options.year;
+  this.speak = function () {
+    console.log(this.make);
```

Comment:
You can put this in one single `console.log` call. Use Template strings here to make this all work on one line.